lib/elixir/lib/map_set.ex

```diff
@@ -171,6 +171,21 @@ defmodule MapSet do
     end
   end
 
+  @doc """
+  Returns a set with elements that are present in only one but not both sets.
+
+  ## Examples
+
+      iex> MapSet.disjoint(MapSet.new([1, 2, 3]), MapSet.new([2, 3, 4]))
+      MapSet.new([1, 4])
+  """
+  @spec disjoint(t(val1), t(val2)) :: t(val1) when val1: value, val2: value
+  def disjoint(left = %MapSet{}, right = %MapSet{}) do
+    diff_left_right = MapSet.difference(left, right)
+    diff_right_left = MapSet.difference(right, left)
+    MapSet.union(diff_left_right, diff_right_left)
```

Comment:
I like this approach because it benefits from the optimizations in `MapSet.difference/2` and `MapSet.union/2` without friction. I wrote a quick [benchmark](https://gist.github.com/v0idpwn/65054671928a168ffed2b7e351c7cfd6) and my implementation using maps directly where possible didn't achieve significant reproducible difference, except maybe for very small inputs.

Reply:
Seems to be significantly quicker!
https://gist.github.com/v0idpwn/c2f2d8103d16d496f6eb03c4a76fa9fe

Reply:
Thank you @v0idpwn!

@scudelletti, can you please implement the version above? Thanks!

Reply:
Thank you both, it seems to work quite well! 👍 

I pushed the change in the last commit.